migration/identity-migration/src/main/java/io/camunda/migration/identity/MigrationRunner.java

```diff
@@ -38,18 +42,37 @@ public void run() {
     final String command =
         args.containsOption("command") ? args.getOptionValues("command").getFirst() : "migrate";
     if (!asList("migrate", "status").contains(command)) {
-      if ("migrate".equals(command)) {
-        authorizationMigrationHandler.migrate();
-      }
       throw new IllegalArgumentException("Unknown command: " + command);
     }
 
-    // TODO: place holder to logic
-    System.out.println("Migration Logic");
+    if ("migrate".equals(command)) {
+      migrate();
+    }
+  }
+
+  private void migrate() {
+    while (true) {
+      try {
+        tenantMigrationHandler.migrate();
+        authorizationMigrationHandler.migrate();
+        break;
+      } catch (final Exception e) {
+        LOGGER.warn("Migration failed, let's retry!", e);
+        try {
+          Thread.sleep(1000);
+        } catch (final InterruptedException ex) {
+          throw new RuntimeException(ex);
+        }
+      }
+    }
   }
 
   @Override
   public void acceptArguments(final ApplicationArguments args) {
     this.args = args;
   }
+
+  public static void main(final String[] args) {
```

Comment:
:x: This should be run as part of the StandaloneMigration, which runs all migrations, or the StandaloneCamunda by enabling the `Profile.IDENTITY_MIGRATION`